doc/contributing/node-api-tooling.md

```diff
@@ -0,0 +1,179 @@
+1. The tooling parses the input file and does several transformations to the AST (Abstract Syntax Tree).
+1. The tooling generates a JSON output that contains the metadata and content of the Markdown file
+1. The tooling generates a HTML output that contains a human-readable and ready to-view version of the file.
+
+These are the summarised steps of the API tooling. On the following sections we're diving deep what each part does, means and how it works.
+
+This documentation mainly serves the purpose of explaining the existing tooling processes, to allow an easier maintenance and evolution of the tooling. It is not meant to be a guide on how to write documentation for Node.js.
+
+#### Vocabulary & Good to Know's
+
+- AST means "Abstract Syntax Tree" and it is a data structure that represents the structure of a certain data format. In our case, the AST is a "graph" representation of the contents of the Markdown file.
+- MDN means "Mozilla Developer Network" and it is a website that contains documentation for web technologies. We use it as a reference for the structure of the documentation.
+- "Stability Index" is an internal concept for the Stability of a given Node.js module. There's no concrete documentation of the "Stability Index" but it goes as:
+  - Stability 1: Experimental. (This module is Experimental)
+  - Stability 0: Deprecated. (This module is Deprecated)
+  - Stability 2: Stable. (This module is Stable)
+  - Stability 3: Legacy. (This module is Legacy)
+- Within Remark YAML snippets `<!-- something -->` are considered HTML nodes, that's because YAML isn't valid Markdown content. (Doesn't abide by the Markdown spec)
+- "New Tooling" references to the (written from-scratch) API build tooling introduced in `nodejs/nodejs.dev` whose might replace the current one from `nodejs/node`
+
+## CLI Arguments
+
+The tooling requires a main `filename` argument and supports extra arguments (some also required) as shown below:
+
+| Argument | Description | Required | Example |
+| -------- | ----------- | -------- | ------- |
+| `--node-version=` | The version of Node.js that is being documented. It defaults to `process.version` which is supplied by Node.js itself | No | v19.0.0 |
+| `--output-directory=` | The directory where the output files will be generated. | Yes | `./out/api/` |
+| `--apilinks=` | This file is used as an index to specify the source file for eac module | No | `./out/doc/api/apilinks.json` |
+| `--versions-file=` | This file is used to specify an index of all previous versions of Node.js. It is used for the Version Navigation on the API docs page. | No | `./out/previous-doc-versions.json` |
+
+**Note.:** the `apilinks` parameter is generated by the Node.js build process (Makefile). And it is a file containing a JSON object.
+
+**Note.:** the `versions-file` parameter is generated by the Node.js build process (Makefile). And it is a file containing a JSON object.
+
+### Basic Usage
```

Comment:
We mighty also want to mention that it is often run using `make doc`? At least that is the way I remember using it.